### DISCUSSION TITLE
WASM (WebAssembly) & JS (JavaScript) Support Formalization

### DISCUSSION THREAD

**Discussion by ylluminate** [vlang/v#11669] [Ideas]
With many ongoing discussions and requests of JavaScript and WebAssembly support for V and the misunderstandings that it doesn't exactly exist yet (or a clear path doesn't yet exist), I believe it is important to have a semi-formalized discussion and presentation of this information for V.

Should we put something on the [main site](https://github.com/vlang/website) and / or within the [README.md](https://github.com/vlang/v/blob/master/README.md)?

There is ongoing discussion [in Discord](https://discordapp.com/channels/592103645835821068/698186021132566598/709432305286119504). And @prime31 has shared [what works](https://discordapp.com/channels/592103645835821068/592106336838352923/705151015548092527) for him:

> for VS Code users this task will get wasm going for you (pretty_c can be slow if you're codebase if over 20 - 30k lines so drop it if you dont need it):
> ```
>         {
>             "label": "v generate c (WebGL)",
>             "type": "shell",
>             "options": {
>                 "cwd": "${fileDirname}"
>             },
>             "command": "v -pretty_c -d webgl -o ${fileBasenameNoExtension}.c ${fileBasename} && cat <(echo \"#include <emscripten.h>\") ${fileBasenameNoExtension}.c > wasm.c",
>             "group": "build",
>             "problemMatcher": []
>         }
> ```

Any thoughts on formalizing this information and making it all more clear and convenient to inbound V learners and contributors?

**Comment by teymour-aldridge** [comment]
Just a note: "JavaSacript" should be spelled "JavaScript"

**Comment by JalonSolov** [comment]
@medvednikov responded somewhere in the thread that it might be possible to have a `-wasm` or `-os wasm` option or something similar, so that v would take care of the boilerplate code for  you.  No need for extra VS Code macros or anything else.  v would also call the C compiler with the proper options so everything would be done.

**Comment by ylluminate** [comment]
Right, I think it's just a matter of someone just trying their hand at implementing this right now @JalonSolov. If I recall correctly there was an easy way to do this at one point long ago (circa last July?), but seems to have gone away.

**Comment by ylluminate** [comment]
Something done by @Zambonifofex, mentioned [here in Discord](https://discordapp.com/channels/592103645835821068/655076213265530900/711903663387574332):
https://github.com/Zambonifofex/v-wasm

**Comment by tomByrer** [comment]
dupe #737

& good job @Zambonifofex !
https://twitter.com/v_language/status/1262450417562157062

**Comment by JalonSolov** [comment]
This is not a dupe of #737.  This issue is for discussing the way to implement #737 or otherwise get wasm support into V.
